show/reboot_cause.py

```diff
@@ -24,6 +23,100 @@ def read_reboot_cause_file():
     return reboot_cause_dict
 
 
+# Function to fetch reboot cause data from database
+def fetch_data_from_db(module_name, fetch_history=False, use_chassis_db=False):
+    prefix = 'REBOOT_CAUSE|'
+    if use_chassis_db:
+        try:
+            rdb = SonicV2Connector(host='redis_chassis.server', port=6380)
+            rdb.connect(rdb.CHASSIS_STATE_DB)
+            table_keys = rdb.keys(rdb.CHASSIS_STATE_DB, prefix+'*')
+        except Exception:
+            return []
+    else:
+        rdb = SonicV2Connector(host='127.0.0.1')
+        rdb.connect(rdb.STATE_DB, False)   # Make one attempt only
+        table_keys = rdb.keys(rdb.STATE_DB, prefix+'*')
+
+    if table_keys is not None:
+        table_keys.sort(reverse=True)
+
+    table = []
+    d = []
+    for tk in table_keys:
+        r = []
+        append = False
+        if use_chassis_db:
+            entry = rdb.get_all(rdb.CHASSIS_STATE_DB, tk)
+        else:
+            entry = rdb.get_all(rdb.STATE_DB, tk)
+
+        if module_name is not None:
+            if 'device' in entry:
+                if module_name != entry['device'] and module_name != "all":
+                    continue
+                if entry['device'] in d and not history:
+                    append = False
+                    continue
+                elif not entry['device'] in d or entry['device'] in d and history:
+                    if not entry['device'] in d:
+                        d.append(entry['device'])
+                        append = True
+            r.append(entry['device'] if 'device' in entry else "SWITCH")
+
+        name = tk.replace(prefix, "")
+        if "|" in name:
+            name = name[:name.rindex('|')] + ''
+        r.append(name)
+        r.append(entry['cause'] if 'cause' in entry else "")
+        r.append(entry['time'] if 'time' in entry else "")
+        r.append(entry['user'] if 'user' in entry else "")
+        if append and not fetch_history:
+            table.append(r)
+        elif fetch_history:
+            r.append(entry['comment'] if 'comment' in entry else "")
+            if module_name is None or module_name == 'all' or module_name.startswith('SWITCH') or \
+               'device' in entry and module_name == entry['device']:
```

Comment:
@rameshraghupathy can you test if these CLIs crash if the DB is empty? are we handling those corner case gracefully?